surya/util/parallel.py

```diff
@@ -1,6 +1,16 @@
-class FakeParallel():
-    def __init__(self, func, *args):
-        self._result = func(*args)
+from concurrent.futures import Future, Executor
 
-    def result(self):
-        return self._result
+class FakeFuture(Future):
```

Comment:
This is a clever way to do it.  But I think it's a little too complex - the Future and Executor class have their own internal logic that isn't necessarily aligned to what we want, and could cause issues down the line if the definitions change.

We could just do this, which is self-contained:

```
class FakeFuture:
    def __init__(self, func, *args, **kwargs):
        self._result = func(*args, **kwargs)

    def result(self):
        return self._result

class FakeExecutor:
    def __init__(self, **kwargs):
        pass

    def __enter__(self):
        return self

    def submit(self, fn, *args, **kwargs):
        return FakeFuture(fn, *args, **kwargs)
```